subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/DefaultHttpSettings.java

```diff
@@ -94,4 +103,44 @@ public boolean verify(String hostname, SSLSession session) {
             return true;
         }
     };
+
+    private static final SslContextFactory ALL_TRUSTING_SSL_CONTEXT_FACTORY = new SslContextFactory() {
+        private final Supplier<SSLContext> SSL_CONTEXT_SUPPLIER = Suppliers.memoize(new Supplier<SSLContext>() {
+            @Override
+            public SSLContext get() {
+                try {
+                    SSLContext sslcontext = SSLContext.getInstance("TLS");
+                    sslcontext.init(null, ALL_TRUSTING_TRUST_MANAGER, null);
+                    return sslcontext;
+                } catch (NoSuchAlgorithmException e) {
+                    throw UncheckedException.throwAsUncheckedException(e);
+                } catch (KeyManagementException e) {
+                    throw UncheckedException.throwAsUncheckedException(e);
```

Comment:
You can simply use `throw new UncheckedException(e)` in these cases. Actually, `throwAsUncheckedException()` is not very useful when you know the type of exception.

Reply:
Done that.